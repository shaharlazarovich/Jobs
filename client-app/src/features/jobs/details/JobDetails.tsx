import React, { useContext, useEffect } from 'react'
import { Grid } from 'semantic-ui-react'
import { observer } from 'mobx-react-lite'
import { RouteComponentProps } from 'react-router'
import LoadingComponent from '../../../app/layout/LoadingComponent'
import JobDetailedHeader from './JobDetailedHeader'
import JobDetailedInfo from './JobDetailedInfo'
import { RootStoreContext } from '../../../app/stores/rootStore'

interface DetailParams {
  id: string
}

const JobDetails: React.FC<RouteComponentProps<DetailParams>> = ({
  match,
  history
}) => {
  const rootStore = useContext(RootStoreContext);
  const {
          job,
          loadJob,
          loadingInitial
        } = rootStore.jobStore; //destructure of parameters
        
  useEffect(() => {
    loadJob(match.params.id);
  }, [loadJob, match.params.id,history])

  if (loadingInitial) return <LoadingComponent content='Loading job...' />
  if (!job) return <h2>Job Not Found</h2>
  return (
    <Grid data-test="component-job-details">
      <Grid.Column width={16}>
        <JobDetailedHeader job={job} />
        <JobDetailedInfo job={job} />
      </Grid.Column>
    </Grid>
  );
}

export default observer(JobDetails)
