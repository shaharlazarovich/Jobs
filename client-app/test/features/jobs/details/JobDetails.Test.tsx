import { ReactWrapper } from "enzyme";
import React from "react";
import { findByTestAttr } from '../../../common/testUtils';
import JobDetails from "../../../../src/features/jobs/details/JobDetails";
import { setupWrapper } from "../../../common/wrapper";
import { Route } from "react-router-dom";
import { match , history } from '../../../common/mocks';

describe('Job Details UniTest', () => {
    let wrapper: ReactWrapper;
    beforeEach(async () => {
        wrapper = await setupWrapper(<Route component={JobDetails} match={match} history={history} />);
    });
    test('renders without error', () => {
        const component = findByTestAttr(wrapper, 'component-job-details');
        expect(component.length).toBe(1);
    });
    test('should match the snapshot', () => {
        const component = findByTestAttr(wrapper, 'component-job-details');
        expect(component).toMatchSnapshot();
    });
})