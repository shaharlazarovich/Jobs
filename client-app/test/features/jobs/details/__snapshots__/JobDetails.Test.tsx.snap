// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Details UniTest should match the snapshot 1`] = `null`;
